pandas/core/arrays/base.py

```diff
@@ -1073,6 +1066,12 @@ def fillna(
         Length: 6, dtype: Int64
         """
         mask = self.isna()
+        if limit is not None and limit < len(self):
+            modify = mask.cumsum() > limit
```

Comment:
`self.isna()` can return an ExtensionArray, which may not implement `cumsum` because we don't require it to implement `_accumulate` (but do require `_reduce`). Would it be okay to require `_accumulate`? Still not sure about the comparison `>` in that case.

Reply:
@jbrockmendel - any thoughts here?

Reply:
I’d be fine with that. Could add a comment to that effect